src/node/node_state.h

```diff
@@ -314,7 +320,35 @@ namespace ccf
           // replaced with a new one initialised with recovered ledger secrets.
           setup_encryptor(network.consensus_type);
 
-          setup_recovery_hook();
+          bool from_snapshot = !args.config.snapshot.empty();
+
+          setup_recovery_hook(from_snapshot);
+
+          // Snapshot generation is disabled until private recovery is complete
+          recovery_snapshot_tx_interval = args.config.snapshot_tx_interval;
+
+          if (from_snapshot)
+          {
+            // Keep a reference to snapshot for private recovery
+            recovery_snapshot =
+              std::make_unique<std::vector<uint8_t>>(args.config.snapshot);
+
+            LOG_DEBUG_FMT(
+              "Deserialising public snapshot ({})", recovery_snapshot->size());
+            auto rc = network.tables->deserialise_snapshot(
```

Comment:
I had to change the interface to `deserialise_snapshot()` to return the `view_history` contained in the snapshot rather than set it on the consensus directly. This is because here, to have a consensus, we need a `State` and to have a `State`, we need to our node ID... which we don't have until we have deserialised the snapshot. We could add setters on the state to set the node ID after having created the consensus but I think this solution is neater.